app/api/v1/habit/create-habit/route.js

```diff
@@ -7,31 +7,47 @@ import connectDB from "@/lib/db/configs/connection";
 
 // responses
 import { badRequest, created, unauthorized } from "@/utils/responses";
+import { findOne, insertOne } from "@/lib/db/repository";
 
 export async function POST(request) {
   try {
     const data = await request.json();
+
     await connectDB();
     const { userId } = auth();
     if (!userId) {
       return unauthorized();
     }
-    const individual = await DB_MODELS.USER.findOne({ clerk_user_id: userId });
-    if (!individual) {
-      return unauthorized();
-    }
-    const new_chart = new DB_MODELS.CHART({
-      name: data.name,
-      description: data.description,
-      user_id: individual._id,
-      icon: data.icon,
-      contributions_per_day: data.contributions_per_day,
-      contribs: [],
-      color: data.color,
+    const [userResult, userResultError] = await findOne({
+      collection: DB_MODELS.USER,
+      query: {
+        clerk_user_id: userId,
+      },
+    });
+    if (userResultError) return internalServerError(userResultError);
+
+    const [newChartResult, newChartResultError] = await insertOne({
+      model: DB_MODELS.CHART,
+      data: {
+        name: data.name,
+        description: data.description,
+        user_id: userResult._id,
+        icon: data.icon,
+        contributions_per_day: data.contributions_per_day,
+        contribs: [],
+        color: data.color,
+      },
+    });
+    if (newChartResultError) return internalServerError(newChartResultError);
+
+    await userResult.updateOne({
+      $push: {
+        charts: newChartResult._id,
+      },
     });
-    await new_chart.save();
     return created();
   } catch (e) {
+    console.log(e);
     return badRequest();
```

Comment:
its not a bad request, its server error, bad req is when user is missing with some data, send server error with some msg in body and code